app/src/main/java/be/huffle/allowanceplanner/ui/allowance/AllowanceViewModel.java

```diff
@@ -0,0 +1,28 @@
+package be.huffle.allowanceplanner.ui.allowance;
+
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MutableLiveData;
+import androidx.lifecycle.ViewModel;
+
+import java.text.DecimalFormat;
+import java.util.Currency;
+import java.util.Locale;
+
+public class AllowanceViewModel extends ViewModel
+{
+	private MutableLiveData<String> mText;
+	private double allowance = 0;
+	private DecimalFormat decimalFormat = new DecimalFormat();
```

Comment:
You could define `mText` and `decimalFormat` as `final`. This would prevent you from accidentally setting its value to something else, and forces you to set its value either in the constructor or directly. This can prevent dumb null pointer exception. (You don't run into those issues here, but it's still nice to know and apply in order to prevent those issue)